source/isaaclab/isaaclab/envs/mdp/events.py

```diff
@@ -1092,6 +1092,94 @@ def reset_joints_by_offset(
+    This function samples random values from the given ranges and sets the end-effector pose of the robot to these values.
+    The end-effector pose is defined by the position and orientation of the end-effector in the world frame.
+    The function then computes the joint positions and velocities that achieve this end-effector pose and sets them
+    into the physics simulation.
+    """
+    # extract the used quantities (to enable type-hinting)
+    asset: Articulation = env.scene[asset_cfg.name]
+
+    end_effector_jacobian_index = asset.find_bodies(ee_frame_name)[0][0] - 1
+    arm_joint_ids = asset.find_joints(arm_joint_names)[0]
+
+    # sample random end-effector pose
+    num_envs = len(env_ids)
+
+    # Fill position
+    target_pos_b = torch.empty((num_envs, 3), device=asset.device)
+    target_pos_b[:, 0] = torch.empty(num_envs, device=asset.device).uniform_(*x_range)
+    target_pos_b[:, 1] = torch.empty(num_envs, device=asset.device).uniform_(*y_range)
+    target_pos_b[:, 2] = torch.empty(num_envs, device=asset.device).uniform_(*z_range)
+
+    solving = torch.ones(len(env_ids), dtype=torch.bool, device=asset.device)
+
+    for index in range(max_inverse_kinematics_iterations):
+        if index == 0:
+            target_joint_pos = asset.data.default_joint_pos.clone()[env_ids][:, arm_joint_ids]
+        else:
+            root_pos_w = asset.data.root_pos_w[env_ids]
+            root_quat_w = asset.data.root_quat_w[env_ids]
+
+            joint_pos = asset.data.joint_pos[env_ids][:, arm_joint_ids]
+            ee_pos_w = asset.data.body_link_pos_w[env_ids][:, end_effector_jacobian_index, :].squeeze(1)
+            ee_pos_b = ee_pos_w - root_pos_w
+
+            position_error = target_pos_b - ee_pos_b
+            pos_error_norm = position_error.norm(dim=-1)
+            solving = pos_error_norm >= position_error_threshold
+            if not solving.any():
+                break
+
+            solving_index = solving.nonzero(as_tuple=False).squeeze(-1)
+            jacobian = asset.root_physx_view.get_jacobians()[env_ids][
+                :, end_effector_jacobian_index, 0:3, arm_joint_ids
+            ][solving_index]
+            base_rot_matrix = math_utils.matrix_from_quat(math_utils.quat_inv(root_quat_w[solving_index]))
+
+            jacobian_base = jacobian.clone()
+            jacobian_base = torch.bmm(base_rot_matrix, jacobian)
+            jacobian_base_transpose = torch.transpose(jacobian_base, dim0=1, dim1=2)
+            lambda_matrix = (lambda_val ** 2) + torch.eye(n=jacobian_base.shape[1], device=asset.device)
+
+            delta_joint_pos = torch.zeros_like(joint_pos, device=asset.device)
+            delta_joint_pos[solving_index] = (
+                jacobian_base_transpose
+                @ torch.inverse(jacobian_base @ jacobian_base_transpose + lambda_matrix)
+                @ position_error[solving_index].unsqueeze(-1)
+            ).squeeze(-1)
```

Comment:
off the top of my head (but didn't think too deeply about this), we could probable rewrite part of this to use `torch.linalg.solve` which would be better to avoid ill-conditioning, but assuming this works consistently, I don't mind